luigi/tasks/genome_mapping/genome_mapping_tasks.py

```diff
@@ -0,0 +1,395 @@
+
+    def output(self):
+        filename = os.path.join(genome_mapping().rnacentral_fasta(self.species), 'rnacentral.fa')
+        return luigi.LocalTarget(filename)
+
+    def records(self):
+        return gm.export_rnacentral_fasta(db(), taxid=self.taxid)
+
+
+class CleanSplitFasta(luigi.Task):
+    """
+    Use seqkit to split RNAcentral species-specific fasta files into chunks to
+    speed up blat searches. Filter out sequences that are too short or too long
+    to be mapped with blat.
+    """
+    taxid = luigi.IntParameter()
+    species = luigi.Parameter()
+    num_seq = luigi.IntParameter(default=1000)
+    min_length = luigi.IntParameter(default=20)
+    max_length = luigi.IntParameter(default=100000)
+    prefix = 'rnacentral-clean.part_'
+
+    def requires(self):
+        return GetFasta(taxid=self.taxid, species=self.species)
+
+    def output(self):
+        chunk_fasta = os.path.join(genome_mapping().chunks(self.species),
+                                   '{}*.fasta'.format(self.prefix))
+        fasta_files = []
+        for filename in iglob(chunk_fasta):
+            fasta_files.append(luigi.LocalTarget(filename))
+        return fasta_files
+
+    def run(self):
+        cmd = ('seqkit seq --min-len {min_length} --max-len {max_length} '
+                   '{fasta} > {temp_dir}/rnacentral-clean.fasta && '
+               'seqkit split --quiet -f -s {num_seq} --out-dir {out_dir} '
+                   '{temp_dir}/rnacentral-clean.fasta && '
+               'rm {temp_dir}/rnacentral-clean.fasta').format(
+               max_length=self.max_length,
+               min_length=self.min_length,
+               fasta=self.input().path,
+               num_seq=self.num_seq,
+               out_dir=genome_mapping().chunks(self.species),
+               temp_dir=genome_mapping().genome(self.species))
+        status = subprocess.call(cmd, shell=True)
+        if status != 0:
+            raise ValueError('Failed to run seqkit: %s' % cmd)
+        if not self.output():
+            filename = os.path.join(genome_mapping().chunks(self.species),
+                                    '{}empty-file.fasta'.format(self.prefix))
+            with open(filename, 'w') as f:
+                f.write(' ')
+
+
+class GetAssemblyInfoJson(luigi.Task):
+    """
+    Download assembly description from Ensembl to get a list of chromosomes.
+    """
+    species = luigi.Parameter()
+    division = luigi.Parameter()
+
+    def output(self):
+        genome_path = genome_mapping().genome(self.species)
+        filename = os.path.join(genome_path, 'ensembl-assembly-info.json')
+        return luigi.LocalTarget(filename)
+
+    def get_url(self):
+        url = 'http://rest.{division}.org/info/assembly/{species}?content-type=application/json'
+        if self.division == 'Ensembl':
+            url = url.format(division='ensembl', species=self.species)
+        else:
+            url = url.format(division='ensemblgenomes', species=self.species)
```

Comment:
Unless we are sure that we will only ever have those two divisions, it would be better to do `elif self.division == 'EnsemblGenoemes ... else raise ValueError("Bad")`.

Reply:
there are 5 more divisions but they are all ensemblgenomes (protists, bacteria etc). One would have to do 5 `elifs` - not sure if it's better